### ISSUE TITLE
Duplicates in ANMN NRS - Maria and North Stradbroke Islands - Near real-time meteorology and sea surface temperature

### ISSUE THREAD

**Issue by dnahodil** [aodn/content#72]
@mhidas this is an issue that has been reported to me by a user just before Chrismas.

<b>What to do :</b>
-On step 1 select the data collection 
IMOS - ANMN National Reference Stations - Maria and North Stradbroke Islands - Near real-time meteorology and sea surface temperature .
-On step 2 filter temporally from 2014/06/12 to 2014/06/13 and choosing site code NRSMAI.
-On step 3 choose download as CSV and then download as un-subsetted netcdf..

<b>What to expect :</b>
We retrieve a CSV file with unique data values per time stamp.
We retrieve the relevant netcdf file.

<b>What happens :</b>
In the CSV, from 2014-06-12T01:06:28 there are duplicates of same values, only the field index is different. These duplicated rows also happen to have their boolean fields (air_temperature_b, etc...) not documented.
We retrieve an empty failed netcdf in addition to the relevant one, like if an old version of the netcdf was still present somewhere and would affect the CSV content?

*Originally reported by @ggalibert as https://github.com/aodn/aodn-portal/issues/1545*

**Comment by dnahodil** [comment]
![05ca4c4e-94fc-11e4-9902-c25e35f21b43](https://cloud.githubusercontent.com/assets/2774917/5657248/ca75dcee-973f-11e4-944d-e3e914e5f6c5.png)

**Comment by mhidas** [comment]
There are two issues here:
  1) The .csv files we get from CSIRO have many duplicate timestamps. I have recently added some processing steps to remove these before creating the netCDF.
  2) We've somehow ended up with an old netCDF file (which still had duplicates) not being deleted from the database when it was deleted from the filesystem. Its data included in the CSV download, while trying to download the non-existent file gives the error in the download log.

So 1) has been fixed, and 2) should not normally occur because the harvester always deletes the data associated with any deleted file. The only possibility I can imagine is that the harvester died during indexing or running liquibase, as these are the only things that happen before handling deleted files. The next time it ran, the indexing would have been re-done and the file was no longer picked up as having just been deleted.

@jonescc Does this make sense? Any way to avoid this?
